tensorflow/core/transforms/remapper/remapping_helper.h

```diff
@@ -0,0 +1,229 @@
+                        const StringRef& attr_name = "T") {
+    if (!lhs_op->hasAttrOfType<TypeAttr>(attr_name) ||
+        !rhs_op->hasAttrOfType<TypeAttr>(attr_name))
+      return false;
+    Type lhs_dtype = lhs_op->getAttrOfType<TypeAttr>(attr_name).getValue();
+    Type rhs_dtype = rhs_op->getAttrOfType<TypeAttr>(attr_name).getValue();
+    return !lhs_dtype.isa<NoneType>() && !rhs_dtype.isa<NoneType>() &&
+           (lhs_dtype == rhs_dtype);
+  }
+
+  // This function is currently used by contraction ops.
+  bool IsGpuCompatibleDataType(Operation* contraction_op,
+                               const StringRef& attr_name = "T") {
+    Type dtype;
+    if (contraction_op->hasAttrOfType<TypeAttr>(attr_name))
+      dtype = contraction_op->getAttrOfType<TypeAttr>(attr_name).getValue();
+    else
+      return false;
+    if (dialect_->IsConv2D(contraction_op))
+      return dtype.isa<Float32Type>();
+    else if (dialect_->IsMatMul(contraction_op))
+      return dtype.isa<Float32Type>() || dtype.isa<Float64Type>();
+    else
+      return false;
+  }
+
+  // This function is currently used by contraction ops.
+  bool IsCpuCompatibleDataType(Operation* contraction_op,
+                               const StringRef& attr_name = "T") {
+    Type dtype;
+    if (contraction_op->hasAttrOfType<TypeAttr>(attr_name))
+      dtype = contraction_op->getAttrOfType<TypeAttr>(attr_name).getValue();
+    else
+      return false;
+    if (is_onednn_enabled_) {
+      // Only contraction ops are supported.
+      bool is_supported = IsContraction(contraction_op) ||
+                          dialect_->IsAnyBatchMatMul(contraction_op);
+      return is_supported &&
+             (dtype.isa<Float32Type>() || dtype.isa<BFloat16Type>());
+    }
+
+    if (dialect_->IsConv2D(contraction_op))
+      return dtype.isa<Float32Type>() || dtype.isa<Float64Type>();
+    else if (dialect_->IsMatMul(contraction_op))
+      return dtype.isa<Float32Type>();
+    else
+      return false;
+  }
+
+  // This function is currently used by convolution type op
+  bool IsGpuCompatibleDataFormat(Operation* conv_op,
+                                 const StringRef& attr_name = "data_format") {
+    StringRef data_format;
+    if (conv_op->hasAttrOfType<StringAttr>(attr_name))
+      data_format = conv_op->getAttrOfType<StringAttr>(attr_name).getValue();
+    else
+      return false;
+    if (dialect_->IsConv2D(conv_op))
+      return data_format == "NHWC" || data_format == "NCHW";
+    else
+      return false;
+  }
+
+  // This function is currently used by convolution type op
+  bool IsCpuCompatibleDataFormat(Operation* conv_op,
+                                 const StringRef& attr_name = "data_format") {
+    StringRef data_format;
+    if (conv_op->hasAttrOfType<StringAttr>(attr_name))
+      data_format = conv_op->getAttrOfType<StringAttr>(attr_name).getValue();
+    else
+      return false;
+    if (dialect_->IsConv2D(conv_op))
```

Comment:
Enclose them with {} will be more clear because it's multiple lines (even thought it's single line statement)